daemon/graphdriver/graphtest/graphtest_unix.go

```diff
@@ -329,8 +332,14 @@ func DriverTestSetQuota(t *testing.T, drivername string) {
 
 	quota := uint64(50 * units.MiB)
 
-	err = writeRandomFile(path.Join(mountPath.Path(), "file"), quota*2)
-	if pathError, ok := err.(*os.PathError); ok && pathError.Err != unix.EDQUOT {
-		t.Fatalf("expect write() to fail with %v, got %v", unix.EDQUOT, err)
+	// Try to write a file smaller than quota, and ensure it works
+	err = writeRandomFile(path.Join(mountPath.Path(), "file"), quota/2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Try to write a file bigger than quota, and ensure it does not work
+	if pathError, ok := err.(*os.PathError); ok && pathError.Err != unix.EDQUOT && pathError.Err != unix.ENOSPC {
```

Comment:
1. Looks like you're missing a line here? Something like
```golang
+	err = writeRandomFile(path.Join(mountPath.Path(), "file"), quota*2)
```

2. Since you have already exhausted half the quota by doing a write above, you might either go with a smaller value than `quota*2`, or remove the file written above.